core-c-bridge/include/temporal-sdk-core-c-bridge.h

```diff
@@ -629,6 +652,130 @@ typedef struct TemporalCoreWorkerReplayPushResult {
   const struct TemporalCoreByteArray *fail;
 } TemporalCoreWorkerReplayPushResult;
 
+/**
+ * Interop function pointer signature of the retry predicate of [CallbackBasedWorkerClientPollOptions]
+ */
+typedef bool (*TemporalCoreCallbackBasedWorkerClientNoRetryPredicate)(void *user_data,
+                                                                      enum TemporalCoreCallbackBasedWorkerClientGRPCCode code);
+
+/**
+ * Interop mirror of [temporal_sdk_core::PollOptions]
+ */
+typedef struct TemporalCoreCallbackBasedWorkerClientPollOptions {
+  /**
+   * always non-null (could be empty)
+   */
+  struct TemporalCoreByteArrayRef task_queue;
+  /**
+   * true if `no_retry_predicate` is valid
+   */
+  bool has_no_retry;
+  /**
+   * Decide short‐circuit GRPC error
+   */
+  TemporalCoreCallbackBasedWorkerClientNoRetryPredicate no_retry_predicate;
+  /**
+   * Opaque context pointer passed back to your predicate thunk
+   */
+  void *no_retry_user_data;
+  /**
+   * true if `timeout_override.is_some()`
+   */
+  bool has_timeout_ms;
+  /**
+   * timeout in milliseconds
+   */
+  uint64_t timeout_ms;
+} TemporalCoreCallbackBasedWorkerClientPollOptions;
+
+/**
+ * Function pointer for RPC callbacks delivering a `(success, failure)`
+ * pair as non-owning `ByteArrayRef`s to Rust.
+ */
+typedef void (*TemporalCoreWorkerClientCallbackTrampoline)(void *user_data,
```

Comment:
```suggestion
typedef void (*TemporalCoreWorkerClientResultCallback)(void *user_data,
```

Pedantic, but we have not traditionally called this a trampoline in this C header, can it just be called a callback?